x-pack/plugins/observability_solution/inventory/public/components/app_root/header_action_menu/share_link.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import copy from 'copy-to-clipboard';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { useKibana } from '../../../hooks/use_kibana';
+
+const SHARE_BUTTON_LABEL = i18n.translate('xpack.inventory.shareLink.shareButtonLabel', {
+  defaultMessage: 'Share',
+});
+
+const SHARE_TOAST_SUCCESS_LABEL = i18n.translate(
+  'xpack.inventory.shareLink.shareToastSuccessLabel',
+  { defaultMessage: 'Short URL copied to clipboard!' }
+);
+
+const SHARE_TOAST_FAILURE_LABEL = i18n.translate(
+  'xpack.inventory.shareLink.shareToastFailureLabel',
+  { defaultMessage: 'Short URL unable to be copied.' }
+);
+
+function useShortUrlService() {
+  const {
+    services: { share, notifications },
+  } = useKibana();
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const copyShortUrl = useCallback(async () => {
+    setIsLoading(true);
+
+    try {
+      const shortUrls = share.url.shortUrls.get(null);
+
+      const { url } = await shortUrls.createFromLongUrl(window.location.toString());
+
+      setIsLoading(false);
+
+      if (copy(url)) {
+        notifications.toasts.addSuccess({
```

Comment:
While this works smoothly: `toggleShareContextMenu` provides this for free https://github.com/elastic/kibana/blob/574af418a2dd30ad19eee0a4c49b82b41c489470/src/plugins/share/public/types.ts#L30 

@Bluefinger have you tried it

Reply:
I have tried it now, but it doesn't seem to be able to give me short urls (even though I have tried making sure it is configured for short urls), which is part of the acceptance criteria. And currently, we are only doing copied URLs, so creating an overlay just to click again to copy a URL is not great UX. We also don't have the export to CSV option at the moment.